### PR TITLE
[AArch64][GlobalISel] Generate libcall for s128 G_FPTRUNC

### PR DESCRIPTION
**Pull request by madhur13490**
This patch would generate libcall if the source type is s128. This is the root cause for many fallbacks in RajaPerf benchmark.

### CODE REVIEW

File: llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp [hunk 1]
```diff
@@ -646,6 +646,7 @@ AArch64LegalizerInfo::AArch64LegalizerInfo(const AArch64Subtarget &ST)
 
   // FP conversions
   getActionDefinitionsBuilder(G_FPTRUNC)
+      .libcallIf(typeIs(1, s128))
```
Comment (tschuett) [inline llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp:649]: I would prefer `legalFor` as the first line.

File: llvm/test/CodeGen/AArch64/fptrunc.ll [hunk 2]
```diff
@@ -32,6 +32,24 @@ entry:
...
+; CHECK-SD-NEXT:    ldr x30, [sp], #16 // 8-byte Folded Reload
+; CHECK-SD-NEXT:    ret
+;
+; CHECK-GI-LABEL: fptrunc_f128_f64:
+; CHECK-GI:       // %bb.0: // %entry
+; CHECK-GI-NEXT:    b __trunctfdf2
+entry:
+  %c = fptrunc fp128 %a to double
+  ret double %c
+}
```
Comment (tschuett) [inline llvm/test/CodeGen/AArch64/fptrunc.ll:51]: Could you add more tests? E.g. to single and half?

File: llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp (hunk 1 above)
Comment (tschuett) [reply llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp:649]: Could it be a list: `.libcall({{s16, s128}, {s32, s128}, {s64, s128}})`.

Comment (madhur13490) [reply llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp]: Yes, that sounds better. It is `libcallFor`

File: llvm/test/CodeGen/AArch64/fptrunc.ll [hunk 3]
```diff
@@ -50,6 +50,41 @@ entry:
...
+; CHECK-SD-NEXT:    ldr x30, [sp], #16 // 8-byte Folded Reload
+; CHECK-SD-NEXT:    ret
+;
+; CHECK-GI-LABEL: fptrunc_f128_f32:
+; CHECK-GI:       // %bb.0: // %entry
+; CHECK-GI-NEXT:    b __trunctfsf2
+entry:
+  %c = fptrunc fp128 %a to float
+  ret float %c
+}
```
Comment (arsenm) [inline llvm/test/CodeGen/AArch64/fptrunc.ll:87]: Test the vector cases too?

File: llvm/test/CodeGen/AArch64/GlobalISel/legalizer-info-validation.mir [hunk 4]
```diff
@@ -1,3 +1,4 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 5
```
Comment (tschuett) [inline llvm/test/CodeGen/AArch64/GlobalISel/legalizer-info-validation.mir:1]: Is this change necessary?

File: llvm/test/CodeGen/AArch64/fptrunc.ll (hunk 3 above)
Comment (madhur13490) [reply llvm/test/CodeGen/AArch64/fptrunc.ll:70]: Done

**Review by aemerson — APPROVED** [review]
